src/shared/components/EditPostModal.tsx

```diff
@@ -0,0 +1,252 @@
+import { FormEvent, useState, ChangeEvent } from "react";
+import {
+  Box,
+  Button,
+  Input,
+  InputBase,
+  Modal,
+  Typography,
+} from "@mui/material";
+import useTheme from "features/theme/useTheme";
+import { Close, Check, InsertPhotoOutlined } from "@mui/icons-material";
+import { getSecureUrl, uploadMedia } from "shared/utils/cloudinaryUtil";
+import { useForm, useImageInput } from "shared/hooks";
+import { BaseCreatePostModel, BaseFeedType } from "shared/types";
+import { useSelector } from "react-redux";
+import { getCurrentUser } from "features/auth/authSlice";
+import axios from "axios";
+
+type ModalProps = {
+  readonly modalName: string;
+  readonly id: number;
+  readonly content?: string;
+  readonly media?: string;
+  readonly likes: { id: number }[];
+  readonly comments: { userId: number }[];
+  readonly handleClose: () => void;
+  readonly handleConfirm: (value: any) => void;
+};
+
+function EditPostModal({
+  modalName,
+  id,
+  content,
+  media,
+  likes,
+  comments,
+  handleClose,
+  handleConfirm,
+}: ModalProps) {
+  const [modalMedia, setModalMedia] = useState(media || null);
```

Comment:
I don't think we need a modalMedia state since you're already using the form hook

Reply:
It turns out we need it because the form component lacks click event handling. When removing an image, we need to change the FormMedia to an empty string to denote it's been removed but I can't do that. There is a button to remove or upload an image and buttons don't have change events so I cannot use the form here to manage the state of media.

Reply:
We can look into this together aswell

Reply:
Gotcha